### PR TITLE
Increase code style lint

### LINKED ISSUE #620: Increase code style lint
### Problem Description

Currently our code is being validated only by `flake8` 'vanilla' and just a few plugins. We would like to increase the code style checks by adding more add-on's that follow our code style and our standards.

Also we would like to ensure that our docstrings are properly written and follow the rest of our format.

### Additional context

We have performed this task already on [RDT](https://github.com/sdv-dev/RDT/) , more precisely on the following issue:
https://github.com/sdv-dev/RDT/issues/248#issuecomment-932085083

### Docstring plugin

We need to add `pydocstyle` plugin with the following lines on our `setup.cfg` file as we are following the [google convention](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).

```
[pydocstyle]
convention = google
add-ignore = D107, D407, D417
```

### Flake8 plugins to be added
Flake8 comes with a lot of different addons that we can use to adapt it to our codestyle and checking, here is a list of plugins that I found to be interesting for us:

* `flake8-builtins` - Check for python builtins being used as variables or parameters.
* `flake8-comprehensions` - Helps you write better list/set/dict comprehensions.
* `flake8-debugger` - Debug statement checker.
* `flake8-variables-names` - Extension that helps to make more readable variables names.
* `Dlint` - Tool for encouraging best coding practices and helping ensure Python code is secure.
* `flake8-mock` - Provides checking mock non-existent methods.
* `flake8-fixme` - Check for FIXME, TODO and other temporary developer notes.
* `flake8-eradicate` - Plugin to find commented out or dead code.
* `flake8-mutable` - Extension for mutable default arguments.
* `flake8-print` - Check for print statements in python files.
* `flake8-pytest-style` - Plugin checking common style issues or inconsistencies.
* `flake8-quotes` - Extension for checking quotes in python.
* `flake8-multiline-containers` - Plugin to ensure a consistent format for multiline containers.
* `pandas-vet` - Plugin that provides opinionated linting for pandas code.
* `pep8-naming` - Check the PEP-8 naming conventions.
* `flake8-expression-complexity` - Plugin to validate expressions complexity.
*  `flake8-sfs` - String formatting.

### PR DESCRIPTION
**Pull request by fealho**
Resolve #620.

### CODE REVIEW

File: sdv/constraints/tabular.py [hunk 1]
```diff
@@ -1109,7 +1112,7 @@ def _validate_inputs(cls, **kwargs):
         errors = []
         try:
             super()._validate_inputs(**kwargs)
-        except MultipleConstraintsErrors as mce:
```
Comment (amontanez24) [inline sdv/constraints/tabular.py:1112]: minor: can we change this to `as agg_error`

File: sdv/tabular/utils.py [hunk 2]
```diff
@@ -154,14 +154,20 @@ def handle_sampling_error(is_tmp_file, output_file_path, sampling_error):
...
+        )
     elif output_file_path is not None:
-        print('Error: Sampling terminated. '
-              f'Partial results are stored in {output_file_path}.')
+        error_msg = (
+            f'Error: Sampling terminated. Partial results are stored in {output_file_path}.'
+        )
 
+    if error_msg:
+        raise type(sampling_error)(error_msg + '\n' + str(sampling_error))
```
Comment (amontanez24) [inline sdv/tabular/utils.py:170]: very minor: can we add extra line here

**Review by amontanez24 — APPROVED** [review]
Some very minor comments but looks good 🎉 :shipit:

File: sdv/constraints/tabular.py (hunk 1 above)
Comment (fealho) [reply sdv/constraints/tabular.py:1112]: Good one!

**Pull request merged**